scripts/imgtool/image.py

```diff
@@ -85,11 +87,12 @@ def get(self):
 class Image():
 
     def __init__(self, version=None, header_size=IMAGE_HEADER_SIZE,
-                 pad_header=False, pad=False, align=1, slot_size=0,
-                 max_sectors=DEFAULT_MAX_SECTORS, overwrite_only=False,
-                 endian="little"):
+                 dependencies={}, pad_header=False, pad=False, align=1,
```

Comment:
I guess it would be better to use

```
dependencies=None,
...
if dependencies is None:
    dependencies = dict()
```
Not sure if this antipattern applies to constructors as well though. But I guess the dict would be appended to by subsequent calls.

Ref https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
Image is only instantiated once, and `create` also only called once, so you could pass it like this, but the main issue seems to be one of design, which is: why are `dependencies` being passed to both the constructor and `create` function? You should decide which one to use...

Reply:
I've updated the script, thank you for the feedback.